packages/devextreme/js/__internal/core/state_manager/state_manager.ts

```diff
@@ -0,0 +1,199 @@
+    });
+
+    this.devToolsConnector.connect();
+
+    this.logger.info('StateManager initialized');
+  }
+
+  trackStateOf(sourceData: StateManagementTypes.StateSource, sourceDataId?: string): void {
+    const preparedSourceDataId = sourceDataId ?? sourceData?.constructor?.name;
+
+    if (!sourceData) {
+      this.logger.error('State source cannot be null or undefined');
+      return;
+    }
+
+    if (this.componentState[preparedSourceDataId]) {
+      this.logger.debug(`State source with ID '${preparedSourceDataId}' is already tracked. Overwriting.`);
+    }
+
+    Object.entries(sourceData).forEach(([propertyName, propertyValue]) => {
+      if (this.hasValueContainerManagerFor(propertyValue)) {
+        if (!this.componentState[preparedSourceDataId]) {
+          this.componentState[preparedSourceDataId] = {};
+        }
+
+        const isObject = propertyValue !== undefined && propertyValue !== null
+          && typeof propertyValue === 'object';
+        this.componentState[preparedSourceDataId][propertyName] = isObject
+          ? new WeakRef(propertyValue) : propertyValue;
+
+        this.trackStateSourceChanges(
+          preparedSourceDataId,
+          propertyName,
+          propertyValue,
+        );
+      } else {
+        this.logger.debug(`No value container manager found for the '${propertyName}' property of the '${preparedSourceDataId}' state source`);
+      }
+    });
+  }
+
+  private trackStateSourceChanges(
+    stateId: string,
+    propertyName: string,
+    propertyValue: StateManagementTypes.MaybeValueContainer,
+  ): void {
+    const valueContainerManager = this.findValueContainerManagerFor(propertyValue);
+
+    if (!valueContainerManager) {
+      this.logger.debug(`No value container manager found for the '${propertyName}' property of the '${stateId}' state`);
+      return;
+    }
+
+    const fullPathToProperty = joinStatePath(stateId, propertyName);
+
+    try {
+      valueContainerManager.trackChanges(
+        (valueContainerChange: StateManagementTypes.ValueContainerChange) => {
+          const valueContainerChangeCopy = {
+            ...valueContainerChange,
+            payload: { ...valueContainerChange.payload, path: fullPathToProperty },
+          };
+
+          const { previousValue, newValue } = valueContainerChange.payload;
+
+          if (typeof previousValue === 'object' && previousValue !== null) {
+            valueContainerChangeCopy.payload.previousValue = deepCopy(previousValue);
+          }
+
+          if (typeof newValue === 'object' && newValue !== null) {
+            valueContainerChangeCopy.payload.newValue = deepCopy(newValue);
+          }
+
+          const updatedComponentState = this.getComponentState();
+
+          this.devToolsConnector
+            .sendAction(
+              'UPDATE',
+              valueContainerChangeCopy.payload,
+              updatedComponentState,
+            );
+        },
+      );
```

Comment:
Let's make it shorter and simpler

```suggestion
      valueContainerManager.trackChanges(
        (valueContainerChange: StateManagementTypes.ValueContainerChange) => {
          const payloadCopy = deepCopy(valueContainerChange.payload);
          const updatedComponentState = this.getComponentState();

          this.devToolsConnector
            .sendAction(
              'UPDATE',
              payloadCopy,
              updatedComponentState,
            );
        },
      );
```

Reply:
The `fullPathToProperty` is lost in the suggestion.

Reply:
Ok, I don't think it's a major comment
So, let's leave it as is